integration-tests/src/main.rs

```diff
@@ -141,6 +161,26 @@ async fn connect_gateway(
     }
 }
 
+async fn close_socket(
+    tx: Arc<Mutex<SplitSink<WebSocketStream<MaybeTlsStream<TcpStream>>, WSMessage>>>,
+    rx: SplitStream<WebSocketStream<MaybeTlsStream<TcpStream>>>,
+    pinger: JoinHandle<Infallible>,
+) -> Result<()> {
+    pinger.abort();
+    pinger.await.ok(); // make sure the tx is no longer referenced by the arc inside the
+                       // task
+    Arc::try_unwrap(tx)
+        // .context("Could not remove tx from Arc")?
+        .expect("Could not remove tx from Arc")
+        .into_inner()
+        .reunite(rx)
+        .context("Could not reunite socket")?
```

Comment:
the socket is [One of Us](https://www.youtube.com/watch?v=r2PZazpE2rw)

Reply:
> the socket is [One of Us](https://www.youtube.com/watch?v=r2PZazpE2rw)

Yo they're doing a giant Among Us co-op let's play? Sick!